File: Strings/1873/1873.c
#include <stdio.h>
#include <string.h>
 
int main() {

  char rajesh[15], sheldon[15];
  char pedra[] = "pedra", papel[] = "papel", tesoura[] = "tesoura", lagarto[] = "lagarto", spock[] = "spock";
  int c, i, comp;

  scanf("%d", &c);
  
  for(i = 0; i < c; i++) {
    
    scanf("%s %s", rajesh, sheldon);
    
    comp = strcmp(rajesh, sheldon);

    if (comp == 0) {
        printf("empate\n");
    } else if ((strcmp(rajesh, tesoura) == 0) && ((strcmp(sheldon, papel) == 0) || (strcmp(sheldon, lagarto) == 0))) {
        printf("rajesh\n");
    } else if ((strcmp(sheldon, tesoura) == 0) && ((strcmp(rajesh, papel) == 0) || (strcmp(rajesh, lagarto) == 0))) {
        printf("sheldon\n");
    } else if ((strcmp(rajesh, papel) == 0) && ((strcmp(sheldon, pedra) == 0) || (strcmp(sheldon, spock) == 0))) {
        printf("rajesh\n");
    } else if ((strcmp(sheldon, papel) == 0) && ((strcmp(rajesh, pedra) == 0) || (strcmp(rajesh, spock) == 0))) {
        printf("sheldon\n");
    } else if ((strcmp(rajesh, pedra) == 0) && ((strcmp(sheldon, lagarto) == 0) || (strcmp(sheldon, tesoura) == 0))) {
        printf("rajesh\n");
    } else if ((strcmp(sheldon, pedra) == 0) && ((strcmp(rajesh, lagarto) == 0) || (strcmp(rajesh, tesoura) == 0))) {
        printf("sheldon\n");
    } else if ((strcmp(rajesh, lagarto) == 0) && ((strcmp(sheldon, spock) == 0) || (strcmp(sheldon, papel) == 0))) {
        printf("rajesh\n");
    } else if ((strcmp(sheldon, lagarto) == 0) && ((strcmp(rajesh, spock) == 0) || (strcmp(rajesh, papel) == 0))) {
        printf("sheldon\n");
    } else if ((strcmp(rajesh, spock) == 0) && ((strcmp(sheldon, tesoura) == 0) || (strcmp(sheldon, pedra) == 0))) {
        printf("rajesh\n");
    } else if ((strcmp(sheldon, spock) == 0) && ((strcmp(rajesh, tesoura) == 0) || (strcmp(rajesh, pedra) == 0))) {
        printf("sheldon\n");
    }
    
}
    
    return 0;
}

File: Python/Strings/1873.py
# Problema 1873 - Beecrowd - Strings - Nível 2

# Criação de um dict em que a chave é uma opção de arma, e seus valores são as armas que a chave ganha.
armaGanhadora = {
    "tesoura": ["papel", "lagarto"],
    "papel": ["pedra", "spock"],
    "pedra": ["lagarto", "tesoura"],
    "lagarto": ["spock", "papel"],
    "spock": ["tesoura", "pedra"],
}

# Recebendo o número de casos teste.
numCasos = int(input())


for i in range(numCasos):

    # Recebendo as armas do Rajesh e Sheldon, respectivamente.
    armaRajesh, armaSheldon = map(str, input().split(" "))

    # Verificando os casos:
    #
    #   - Se as armas são iguais primeiro.
    #   - Se não, consulte no dict se a arma do Rajesh ganha da arma do Sheldon, usando o atributo ".count"
    #   de listas que retorna quantas ocorrências tem o parâmetro passado. Caso retorne 0, então sabemos que
    #   o Sheldon venceu, pois a arma do Rajesh não ganha da dele.
    #
    # E já apresenta o resultado do jogo.
    if armaSheldon == armaRajesh:
        print("empate")
    elif armaGanhadora[armaRajesh].count(armaSheldon) == 0:
        print("sheldon")
    else:
        print("rajesh")
